bolts.lic

```diff
@@ -0,0 +1,160 @@
+
+    args = parse_args(arg_definitions)
+
+    if args.continue
+      wait_for_script_to_complete('tinker', ['continue', 'bolts'])
+      exit_tasks
+      exit
+    end
+
+    @type = args.type
+    @lumber = args.lumber
+    @tools = args.tools
+
+    get_tools if @tools
+    order_lumber
+    check_flights
+    check_glue
+    make_shafts
+    make_bolts
+    wait_for_script_to_complete('tinker', ['stow', '7', "#{@type} bolts", 'shafts', 'bolts'])
+    exit_tasks
+  end
+
+  def exit_tasks
+    clean_up
+    store_tools if @tools
+  end
+
+  def get_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'get'])
+  end
+
+  def store_tools
+    wait_for_script_to_complete('clerk-tools', ['engineering', 'store'])
+  end
+
+  def order_lumber
+    return if exists?("#{@lumber} lumber") || exists?('shafts')
+    wood = find_piece("#{@lumber} lumber")
+    2.times do
+      order_item(@hometown_data['stock-room'], wood)
+      bput("get #{@lumber} lumber from #{@bag}", "You get", "What were")
+      bput("combine my lumber", "You combine", "You must")
+      bput("stow lumber", "You put")
+    end
+  end
+
+  def check_flights
+    return if exists?('flights')
+    flight = find_piece('bolt flights')
+    2.times do
+      order_item(@hometown_data['stock-room'], flight)
+      bput("stow flights", "You put")
+    end
+  end
+
+  def check_glue
+    if search?('glue')
+      bput('get my glue', 'You get')
+      /(\d+)/ =~ bput('count my glue', 'The wood glue has *\d+ uses remaining')
+      if Regexp.last_match(1).to_i < 5
+        stow_item('glue')
+        dispose('glue')
+        order_item(info['tool-room'], info['glue-number'])
+      end
+    else
+      order_item(info['tool-room'], info['glue-number'])
+    end
+    stow_item('glue')
+  end
+
+  def make_shafts
+    return if exists?('shafts')
+    get_item('shaper')
+    bput("get my #{@lumber} lumber from my #{@bag}", "You get", "What were")
+    bput("shape lumber into bolt shaft", 'Roundtime')
+    stow_item('shaper')
+    bput("stow my shafts in my #{@bag}", "You put")
+  end
+
+  def make_bolts
+    return if exists?('boltheads')
+    bolt_head = { "cougar-claw" => "curved claw", "boar-tusk" => "boar tusk", "sabertooth" => "ivory fang", "angiswaerd" => "angiswaerd tooth", "hele'la" => "needlesharp tooth", "basilisk" => "basilisk fang", "elsralael" => "elsralael tooth", "soot-stained" => "soot-streaked fang", "ice-adder" => "adder fang", "jagged-horn" => "jagged horn", "drake-fang" => "drake fang" }
+    new_type = bolt_head[@type]
+    case bput("get #{new_type} from my #{@bag}", 'You get', 'I could not', 'What were')
+    when /You get/
+      get_item('shaper')
+      bput("shape #{new_type} into bolthead", 'Roundtime', 'You fumble')
+      #bput("combine", "You combine") if bput("Get boltheads from my #{@bag}", "You get", "What were") =~ /You get/
+      stow_item('boltheads')
+    when /I could not/, /What were/
+      beep
+      echo("***Out of #{@type}!***")
```

Comment:
For warnings, consider `DRC.message` instead of `echo`

Reply:
Changed.